axum/src/routing/method_filter.rs

```diff
@@ -21,3 +23,21 @@ bitflags! {
         const TRACE =   0b100000000;
     }
 }
+
+impl TryFrom<Method> for MethodFilter {
+    type Error = Error;
```

Comment:
I don't think we should use `axum::Error` here. Its meant as an opaque error that we use wrap 3rd party errors in. I think we should make a new error type instead. Maybe `UnsupportedMethod` but that sounds quite broad 🤔 wdyt?

Reply:
> Where do I define it ? right there where the macro & trait impl is ?

Yeah I think thats fine. Keeping the code close together makes sense here.

> Can I just do a pub struct `UnsupportedMethod(String)` with a new() constructor ?

I'd do

```rust
#[derive(Debug)]
struct UnsupportedMethod {
    // this is private since users shouldn't be allowed to construct this type.
    // only axum should be able to do that
    // we also don't want a constructor for the same reason
    method: http::Method,
}

impl UnsupportedMethod {
    // should probably provide an accessor though
    fn method(&self) -> &http::Method { ... }
}

impl fmt::Display for UnsupportedMethod {
    // return some readable error in `fmt`
}

// don't need to override source because this isn't wrapper other errors
impl std::error::Error for UnsupportedMethod {}
```